test/common/common/BUILD

```diff
@@ -30,6 +30,18 @@ envoy_cc_test(
     ],
 )
 
+envoy_cc_test(
+    # Same as assert_test above, but with ENVOY_LOG_DEBUG_ASSERT_IN_RELEASE
```

Comment:
https://github.com/envoyproxy/envoy/blob/3840db138734fa4e372db959cd46de5b7b446f3a/ci/do_ci.sh#L308 

did this suffice?

Reply:
It helps and complements testing in an important way by ensuring that all tests run fine with asserts enabled.  this version allows us to detect issues by simply running local tests in opt mode and is very low cost since it doesn't require a full recompile of everything.

Reply:
seems good, not opposed. I hate remembering to running tests with added flags.